Tests/Service/Request/RequestFactoryTest.php

```diff
@@ -160,12 +160,110 @@ public function testBuildFlow()
         );
     }
 
+    /**
+     * @return void
+     */
+    public function testBuildFlowWithQueryParams()
+    {
+        $baseUrl = 'baseUrl';
+        $routeString = '/routeString?param={param}';
+        $originParamValue = 'value with whitespaces';
+        $encodedParamValue = 'value+with+whitespaces';
+        $requestMethod = 'GET';
+        $requestBody = '{requestBody:requestBody}';
+        $endpointProphecy = $this->prophesize(EndpointInterface::class);
+        $endpointProphecy->__call('getBaseUrl', [])
+            ->willReturn($baseUrl)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getPath', [])
+            ->willReturn($routeString)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getMethod', [])
+            ->willReturn($requestMethod)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getRequestFormat', [])
+            ->willReturn(EndpointInterface::FORMAT_JSON)
+            ->shouldBeCalled()
+        ;
+        $endpoint = $endpointProphecy->reveal();
+
+        $uri = $this->prophesize(UriInterface::class)->reveal();
+        $request = $this->prophesize(RequestInterface::class)->reveal();
+
+        $this->endpointRegistryProphecy
+            ->__call('getEndpoint', [Argument::type(ServiceRequestInterface::class)])
+            ->willReturn($endpoint)
+            ->shouldBeCalled()
+        ;
+
+        $this->serializerProphecy
+            ->__call('serialize', [Argument::type(ServiceRequestInterface::class), EndpointInterface::FORMAT_JSON])
+            ->willReturn($requestBody)
+            ->shouldBeCalled()
+        ;
+
+        $converterRoute = str_replace('{param}', $encodedParamValue, $routeString);
+        $this->uriFactoryProphecy
+            ->__call('createUri', [$baseUrl.$converterRoute])
```

Comment:
Why not specify that the argument might be anything? What you really need to check, is that you final URL is properly formatted and encoded.

Reply:
Encoding happens before the `createUri` call and `uriFactory` is just a mock, so I predefine what output `uriFactory` will give after the call, what's the point to check what I predefined?

Reply:
I think in case of we use prophecy in order to ensure the call was performed, we also  could check both input and output values basically. Otherwise how could we be ensured that we call the method with proper data?

Reply:
We can check input and/or output if we can define it as a constant. But we should not add logic and produce the input in the test.

Reply:
when we define a mock, we also define input parameters and if they are different there will be an error in the test. 
Here, in my opinion, more important to check the input parameter since it's the only part of the test I cannot mock or predefine and it's the critical change. 
And what logic do you mean? Concatenating 2 strings? I can replace it with constant, no problem

Reply:
Please do replace it.